frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -244,37 +244,48 @@ export function getMappingOptionByTarget(
   }
 
   const { query, columns } = getParameterColumns(question, parameter);
-  const stageIndexes = _.uniq(columns.map(({ stageIndex }) => stageIndex));
```

Comment:
The previous implementation was incorrect. It tried to match the field ref from the parameter target with mapping options for all stages in the query, ignoring the stage index set in the target. We need to match the parameter target with mapping options for the same stage only. Doing otherwise leads to incorrect matches.